pkg/pquerier/select.go

```diff
@@ -172,19 +172,44 @@ func (queryCtx *selectQueryContext) queryPartition(partition *partmgr.DBPartitio
 
 		newQuery := &partQuery{mint: mint, maxt: maxt, partition: partition, step: step}
 		if aggregationParams != nil {
-			newQuery.preAggregated = aggregationParams.CanAggregate(partition.AggrType())
-			if newQuery.preAggregated || !queryCtx.queryParams.disableClientAggr {
+			newQuery.useServerSideAggregates = aggregationParams.CanAggregate(partition.AggrType())
+			if newQuery.useServerSideAggregates || !queryCtx.queryParams.disableClientAggr {
 				newQuery.aggregationParams = aggregationParams
 			}
 		}
 
-		err = newQuery.getItems(queryCtx, metric, requestAggregatesAndRaw)
+		groupBy := queryCtx.groupBy(partition)
```

Comment:
i would change the name of the function&variable to something more clear, since `groupBy` by default is the "group by" statement you pass in the select but here it's a bit different meaning

Reply:
`groupByLabels`?

Reply:
`parsePreAggregatedLabelGroups` ?
and the variable `preAggregatedLabelGroups`